dotnet/samples/Concepts/Agents/ChatCompletion_Rag.cs

```diff
@@ -0,0 +1,183 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Azure.AI.OpenAI;
+using Azure.Identity;
+using Microsoft.Extensions.AI;
+using Microsoft.Extensions.VectorData;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Agents;
+using Microsoft.SemanticKernel.Connectors.InMemory;
+using Microsoft.SemanticKernel.Data;
+
+namespace Agents;
+
+#pragma warning disable SKEXP0130 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+
+/// <summary>
+/// Demonstrate creation of <see cref="ChatCompletionAgent"/> and
+/// adding simple retrieval augmented generation (RAG) capabilities to it.
+/// </summary>
+/// <remarks>
+/// This example shows how to use the <see cref="TextSearchStore{TKey}"/> class which is designed
+/// to simplify the process of storing and searching text documents by having a built in schema.
+/// If you want to control the schema yourself, you can use an implementation of <see cref="IVectorStoreRecordCollection{TKey, TRecord}"/>
+/// with the <see cref="VectorStoreTextSearch{TRecord}"/> class instead.
+/// </remarks>
+public class ChatCompletion_Rag(ITestOutputHelper output) : BaseTest(output)
```

Comment:
ChatCompletion can be confusing here as we have `ChatCompletion` folder, as we will be moving soon away from the `IChatCompletionService` name to `IChatClient`, suggest just keeping with `Chat` for now. Since RAG is a well known abbrev. I would keep it `Uppercase`

```suggestion
public class ChatAgent_RAG(ITestOutputHelper output) : BaseTest(output)
```